app/views/static/generic_object/custom-image-component.html.haml

```diff
@@ -0,0 +1,43 @@
+.form-group{"ng-show" => "vm.newRecord || vm.changeImage || vm.pictureUrlPath === '' || vm.pictureRemove",
+            "ng-class" => "{'has-error': vm.angularForm.generic_object_definition_image_file_status.$invalid}"}
+  %label.col-md-2.control-label{"for" => "generic_object_definition_image_file_status"}
+    = _("Custom Image File")
+  .col-md-4
+    %input.form-control{"type"        => "file",
+                        "class"       => "filestyle",
+                        "id"          => "generic_object_definition_image_file",
+                        "name"        => "generic_object_definition_image_file"}
+    %input.form-control{"type"        => "hidden",
+                        "id"          => "generic_object_definition_image_file_status",
+                        "ng-model"    => "vm.imageUploadStatus",
+                        "name"        => "generic_object_definition_image_file_status"}
+    %span.help-block
+      {{vm.imageUploadStatus}}
+
+  .col-md-4
+    %miq-button{:name        => t = _("Upload chosen File"),
+                  :title     => t,
+                  :alt       => t,
+                  :enabled   => "true",
+                  'on-click' => "vm.uploadClicked(angularForm)"}
+.form-group{'ng-if' => "vm.pictureUrlPath !== '' && !vm.changeImage && !vm.pictureRemove"}
+  %label.col-md-2.control-label{"for" => "custom_image"}
+    = _("Current Custom Image File")
+  .col-md-4
+    %img.form-control{'ng-src' => "{{vm.pictureUrlPath}}", :style => "width:100px; height:100px;"}
+      .btn.btn-default{:alt => t = _("Remove Custom Image"), :title => t, "ng-click" => "vm.removeImage()"}
+        %i.pficon.pficon-delete
+.form-group{'ng-if' => "vm.pictureUrlPath !== '' && !vm.pictureRemove"}
+  %label.col-md-2.control-label{"for" => "update_image"}
+    = _("Update Current Image")
+  .col-md-4
+    %input.form-control{"type"      => "checkbox",
+                        "id"        => "update_image",
+                        "name"      => "update_image",
+                        "bs-switch" => "",
+                        :data       => {:on_text => _('Yes'), :off_text => _('No')},
+                        "ng-change" => "vm.changeImageSelected()",
+                        "ng-model"  => "vm.changeImage"}
+
+:javascript
+  $(":file").filestyle({placeholder: _("No file chosen"), icon: false});
```

Comment:
`__` ? This is js..

Reply:
But it's a JS inside a haml. So do we still need the `__`?

Reply:
Well, it can be either `placeholder: __("No file chosen"),` or `placeholder: "#{_("No file chosen")}",`

Reply:
Addressed in fb22a12